component/web/security/src/main/java/org/exoplatform/web/security/security/CookieTokenService.java

```diff
@@ -135,34 +77,29 @@ public void start() {
         super.start();
     }
     
-    public String createToken(final Credentials credentials) {
-        return createToken(credentials,"");
+    public String createToken(String username) {
+        return createToken(username,"");
     }
     
-    public String createToken(final Credentials credentials, String type) {
+    public String createToken(String username, String type) {
         if (validityMillis < 0) {
             throw new IllegalArgumentException();
         }
-        if (credentials == null) {
+        if (username == null) {
             throw new NullPointerException();
         }
 
         String cookieTokenString = null;
         while (cookieTokenString == null) {
-            String randomString = nextTokenId();
-            String id = nextRandom();
-            cookieTokenString = new CookieToken(id, randomString).toString();
+            String selector = nextTokenId(); //9 bytes selector
+            String validator = nextRandom(); //9 bytes validator
 
-            String hashedRandomString = hashToken(type+SEPARATOR_CHAR+randomString);
+            String hashedRandomString = hashToken(validator+SEPARATOR_CHAR+type);
             long expirationTimeMillis = System.currentTimeMillis() + validityMillis;
-
-            /* the symmetric encryption happens here */
-            String encryptedPassword = codec.encode(credentials.getPassword());
-            Credentials encodedCredentials = new Credentials(credentials.getUsername(), encryptedPassword);
-
+            cookieTokenString=selector+SEPARATOR_CHAR+validator;
             try {
-                this.tokenStore.saveToken(new GateInTokenStore.TokenData(id, hashedRandomString, encodedCredentials, new Date(expirationTimeMillis), type));
-//                tokenContainer.saveToken(context.getSession(), id, hashedRandomString, encodedCredentials, new Date(expirationTimeMillis, tokenType));
+                this.tokenStore.saveToken(new GateInTokenStore.TokenData(selector, hashedRandomString,
```

Comment:
A security layer has been removed here when removing the usage of `CodecInitializer` to encrypt part of the content. In fact, we 'll need to preserve the usage of `CodecInitializer` to encrypt the stored Token in database.
In this case, when someone accesses the database, he still can't access users accounts until he get access to the `codecKey.txt` file from server.

The `hashedRandomString` has to be encrypted as well IMO and never add it in cookie or expose it as plain text. (it's similar to the password itself).